spec/unit/confine/true_spec.rb

```diff
@@ -12,6 +12,33 @@
     expect { Puppet::Confine::True.new }.to raise_error(ArgumentError)
   end
 
+  describe "when passing in a lambda as a value for lazy evaluation" do
+    it "should accept it" do
+      confine = Puppet::Confine::True.new(lambda { true })
+      expect(confine.values).to eql([true])
+    end
+
+    describe "when enforcing cache-positive behavior" do
```

Comment:
I could also use a lambda that raises an exception if evaluated more than once to test this, but figured it's simpler to just check the instance variable.

Reply:
Making sure that a lambda that raises doesn't cause a transaction to abort in weird ways would definitely be good, but that test might not fit here (depending on how you decide to go about it).